src/core/utils/index.js

```diff
@@ -376,3 +376,26 @@ export const findFileForPlatform = ({ userAgentInfo, platformFiles }) => {
   const platform = agentOsName && userAgentOSToPlatform[agentOsName];
   return (platform && platformFiles[platform]) || platformFiles[OS_ALL];
 };
+
+export const getLocalizedTextWithLinkParts = ({
+  i18n,
+  text,
+  linkStart = '__LINK__',
```

Comment:
This is close, but not what I was thinking for these arguments. The idea I had was this:
- The string you pass in must have markers in it to indicate the start and end of the link.
- In your case those markers are called `linkStart` and `linkEnd`, but really this function should be able to work with any string regardless of what the markers are called.
- So you should be able to pass in the names of the markers into this function (and they can default to `linkStart` and `linkEnd`) so that the function can work on any string.

As an aside, there's not really any reason to allow the caller to configure the value of the substitution (`'__LINK__'` in this case), because that is entirely internal to this function.

Note that I tried to experiment with implementing this myself and I had trouble getting `sprintf` to recognize my marker names. I'm not sure why, but I'll leave that with you to see if you can get it to work.

Reply:
yeh I guess I can remove those as args since this shouldn't change. 

As far as the linkStart and linkEnds names, these need to match since we `split` the string on this to make the array. I used `__Link__` because I thought that was an appropriate name. It could be anything else as long as the name matches otherwise we would have to run split 2 times so I think it makes sense to have the same string name.

Reply:
I'm not sure you understood my comment. Yes, it makes sense for the substitution string to be the same for `linkStart` and `linkEnd` - I wasn't commenting on that. It was about the fact that the markers that `sprintf` should look for should be configurable. So, for example, if your string looks like this: `'Explore more %(myLinkStart)spassword manager%(myLinkEnd)s staff picks.'` it will still work. In that case you'd have to tell the function that the marker for linkStart is `myLinkStart` and the marker for linkEnd is `myLinkEnd`. Does that make sense?

Reply:
So I re-read your comment above and I got it so the linkStart and linkEnd names could be changed. Then I use a similar approach as I had (with __LINK__) so I could split the string as mentioned above.

Reply:
yep, I got it now:) I've updated.